cmake/modules/packages/FindOGDI.cmake

```diff
@@ -12,28 +12,50 @@
 #    OGDI_VERSION
 #
 
-find_path(OGDI_INCLUDE_DIR ecs.h PATH_SUFFIXES ogdi)
+find_package(PkgConfig QUIET)
+if(PKG_CONFIG_FOUND)
+  pkg_check_modules(PC_OGDI QUIET OGDI)
+  set(OGDI_VERSION_STRING ${PC_OGDI_VERSION})
+  set(OGDI_INCLUDE_DIRS ${PC_OGDI_INCLUDE_DIRS})
+endif()
+
+find_path(OGDI_INCLUDE_DIR
+          NAMES ecs.h rpc/rpc.h
```

Comment:
do we need to look for rpc/rpc.h ? This is an implementation detail. GDAL just includes ecs.h

Reply:
Possibly. The original error was

[100%] Building CXX object ogr/ogrsf_frmts/ogdi/CMakeFiles/ogr_OGDI.dir/ogrogdidatasource.cpp.o
In file included from /home/werdna/git/gdal/ogr/ogrsf_frmts/ogdi/ogrogdidatasource.cpp:30:
In file included from /home/werdna/git/gdal/ogr/ogrsf_frmts/ogdi/ogrogdi.h:40:
/usr/include/ogdi/ecs.h:9:10: fatal error: 'rpc/rpc.h' file not found
#include <rpc/rpc.h>
         ^~~~~~~~~~~
1 error generated.

The actual change in Ubuntu is that there are now two transport-independent RPC libraries  tirpc and ntirpc, with headers moved from /usr/include/rpc/ to /usr/include/tirpc/rpc/ and /usr/include/ntirpc/rpc/
It is not clear to me that without pkgconfig, cmake would find rpc/rpc.h

How do I generate a non-pkgconfig, cmake build ?

Reply:
> It is not clear to me that without pkgconfig, cmake would find rpc/rpc.h

indeed, but I suspect that OGDI_INCLUDE_DIR only includes the directory of the first file found, here ecs.h ? (genuinely asking, as I don't have the target env. immediately available)

> How do I generate a non-pkgconfig, cmake build ?

I would just rename /usr/share/pkgconfig/ogdi.pc to ogdi.pc.disabled. But is there a ogdi.pc file at all ??